File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/posIntegrations/PosIntegration.ts
export interface IntegrationFeatures {
    readonly allowsEscPosInvoices: boolean;
    readonly allowsOpeningSessions: boolean;
    readonly allowsAddingItemsToSession: boolean;
    readonly allowsRemovingItemsFromSession: boolean;
}

export enum IntegrationType {
    QuiviViaFacturalusa = 0,
}

export enum PosSyncState {
    SyncFailure = -3,
    PoSOffline = -2,
    Stopped = -1,
    Unknown = 0,
    Running = 1
}

export interface PosIntegration {
    readonly id: string;
    readonly type: IntegrationType;
    readonly isActive: boolean;
    readonly isDianosticErrorsMuted: boolean;
    readonly syncState: PosSyncState;
    readonly features: IntegrationFeatures;
    readonly settings: Record<string, any>;
}

File: Quivi.Backoffice.Api/Dtos/PosIntegration.cs
using Quivi.Domain.Entities.Pos;

namespace Quivi.Backoffice.Api.Dtos
{
    public class IntegrationFeatures
    {
        public bool AllowsEscPosInvoices { get; init; }
        public bool AllowsOpeningSessions { get; init; }
        public bool AllowsAddingItemsToSession { get; init; }
        public bool AllowsRemovingItemsFromSession { get; init; }
    }

    public class PosIntegration
    {
        public required string Id { get; init; }
        public IntegrationType Type { get; init; }
        public bool IsActive { get; init; }
        public SyncState SyncState { get; init; }
        public bool IsDianosticErrorsMuted { get; init; }
        public required IReadOnlyDictionary<IntegrationType, object> Settings { get; init; }
        public required IntegrationFeatures Features { get; init; }
    }
}